[CS2113T-F12-4] CCAManager

ianwangeeen opened the pull request:
This product is designed for the NUS CCA manager who has a lot of administrative tasks to keep track of. The interface uses CLI instead of GUI to fit the preferences of the CCA manager.

izdiyadfrhn committed:
Merge remote-tracking branch 'origin/izdiyad-v2.0' into izdiyad-v2.0

izdiyadfrhn committed:
Update Parser.java

kengjit commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,151 @@
+* `getKeywordStatus()`: Gets the keyword value from the Keyword enum class.
+* `deleteMember(members, entry)`: Deletes the member from the memberList.
+* `deleteMemeber(memberNumber)`: Deletes the member using the index number provided by the user.
+* `writeMemberFile()`: Writes the data to the csv file.
```
Perhaps it would be better to choose a better title than "Note"?

kengjit replied:
Could perhaps split these methods into different components? (i.e. Member, MemberStorage...)

praj-bellakka commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,151 @@
+* [Getting Started](#getting-started)
+* [Design](#design)
+  * [Architecture](#architecture)
+* [Appendix: Requirements](#appendix-requirements)
```
Should there be a  link to the Manual Testing section for consistency?

kengjit commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,151 @@
+* stores the `AttendanceList` object in a csv file with the help of `AttendanceStorage`.
 
-{Describe the value proposition: what problem does it solve?}
+###[Proposed] Storage component 
```
You might want to add a whitespace between the ### and [Proposed] to prevent formatting problems

praj-bellakka replied:
Perhaps your group missed out a space here, as the heading formatting is not showing up
```suggestion
### [Proposed] Storage component 
```

praj-bellakka commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,151 @@
+
+The *sequence diagram* below shows how various components of the architecture interact with one another when a user inputs a **valid** command `"add /m Bob /s A01231234B /p 98765432"`
+
+![Architecture Sequence Diagram](images/ArchitectureSequence.png)
```
Should there be an 'X' to denote the end of the lifeline of an object to show its deletion? If PUML doesn't allow it, maybe note could be written at the start to explain this behaviour.

kengjit replied:
Perhaps you could standardize what you want User to be.

![image](https://user-images.githubusercontent.com/69455772/139000002-8699b0c1-30cd-45b5-ba36-0bf62ec008b5.png)
![image](https://user-images.githubusercontent.com/69455772/139000277-ec099f49-9f71-4fac-adcf-0573f5c21296.png)

andrewtkh1 replied:
Since the `makeMemberEntry()` is creating a new member, should it be pointing to the `Member` box itself as though you are creating a new object? (Example in image below)

![image](https://user-images.githubusercontent.com/69303287/139191631-b5e56341-54ca-4b2e-a829-b18a6050d505.png)

andrewtkh1 replied:
Perhaps it would be better to crop the Sequence diagram to omit the bottom half where the class name is repeated?


e.g) 

![image](https://user-images.githubusercontent.com/69303287/139194118-f51ca8d9-3377-47d0-b7fb-8c6cced7033b.png)

praj-bellakka commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,151 @@
+
+The `training schedule` component
+*can save trainings schedules and read them back into the `TrainingList` object. 
+*automatically deletes in the file whenever a training schedule is deleted from the `TrainingList` object
```
Perhaps you are missing out on a space between the starting character and the star (*) to show the lines as a list
```suggestion
* can save attendance data in CSV format and read them back into the `AttendanceList` object.
* automatically deletes to AttendanceStorage.csv whenever a member is deleted from the `AttendanceList` object.

The `training schedule` component
* can save trainings schedules and read them back into the `TrainingList` object. 
* automatically deletes in the file whenever a training schedule is deleted from the `TrainingList` object
```

kengjit reviewed (commented):
Overall well-done DG, perhaps you could further elaborate on the existing methods and classes

praj-bellakka reviewed (commented):
Looks good so far, but there are some formatting issues and minor fixes that can be taken care of!

Teckwhye committed:
Add delete by name for member

Teckwhye committed:
Merge pull request #121 from Teckwhye/TeckHwee-v2.0

Add delete by name for member

Teckwhye committed:
fix coding style

Teckwhye committed:
Fix expected output file

Teckwhye committed:
Merge pull request #122 from Teckwhye/TeckHwee-v2.0

fix coding style

Teckwhye committed:
Merge branch 'master' into teckhwee-v2.0

Teckwhye committed:
Merge pull request #117 from AY2122S1-CS2113T-F12-4/teckhwee-v2.0

Teckhwee v2.0

Teckwhye committed:
Fix coding style

Teckwhye committed:
Fix coding style

Teckwhye committed:
Merge pull request #123 from Teckwhye/TeckHwee-v2.0

Fix Gradle

Teckwhye committed:
Merge pull request #124 from AY2122S1-CS2113T-F12-4/teckhwee-v2.0

Teckhwee v2.0

Teckwhye committed:
Merge branch 'master' into izdiyad-v2.0

Teckwhye committed:
Merge pull request #120 from izdiyadfrhn/izdiyad-v2.0

Izdiyad v2.0

Teckwhye committed:
Update UserGuide.md

Change information regarding delete member command and expected output for member commands

Teckwhye committed:
Added index to members

Teckwhye committed:
Change expected output.txt

Teckwhye committed:
Merge pull request #125 from Teckwhye/TeckHwee-v2.0

Teck hwee v2.0

Teckwhye committed:
Merge pull request #126 from AY2122S1-CS2113T-F12-4/teckhwee-v2.0

Teckhwee v2.0

Commit:
Add find training function, add Index to TrainingSchedule

Commit:
fix codingstyle error

Commit:
fix traininglisttest

glenn-tck committed:
Merge pull request #127 from poppolette/glenn-v2.0

Add find training function, add Index to TrainingSchedule

Teckwhye committed:
Fix edit member issue

Teckwhye committed:
Merge pull request #128 from Teckwhye/TeckHwee-v2.0

Fix edit member issue

Teckwhye committed:
Fix edit member

Teckhwee v2.0

ianwangeeen committed:
fix minor bug AttendanceStorage

ianwangeeen committed:
fix minor bug AttendanceStorage

fix minor bug AttendanceStorage

andrewtkh1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,151 @@
-## Non-Functional Requirements
 
-{Give non-functional requirements}
+## Appendix: Manual Testing
```
Perhaps can include some test cases in the `manual testing` such that the developer can do to cover all the necessary functions?

![image](https://user-images.githubusercontent.com/69303287/139191446-fe78f211-ad88-4030-a6c7-9937af08d9bc.png)

andrewtkh1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,151 @@
+* `writeMemberFile()`: Writes the data to the csv file.
+
+The *sequence diagram* below shows how various components of the architecture interact with one another when a user inputs a **valid** command `"delete /m 1"`
+![Architecture Sequence Diagram2](images/deleteMemberArchitecture.PNG)
```
Would it be better if both sequence diagrams are consistent in the way that they model the user with the same icon?

![image](https://user-images.githubusercontent.com/69303287/139191896-0b316f01-ebf0-47e7-be81-aa01632401c2.png)

andrewtkh1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,151 @@
+**API** : [`Attendance.java`](https://github.com/AY2122S1-CS2113T-F12-4/tp/blob/master/src/main/java/seedu/duke/attendance/Attendance.java)
 
-{Describe the target user profile}
+![Attendance Component Diagram](images/AttendanceComponent.png)
```
Should the arrow pointing to `Attendance` be a solid arrow instead since there is the `AttendanceList` class has the attribute `Attendance`?

![image](https://user-images.githubusercontent.com/69303287/139192851-900ca1c5-820a-49a5-b7d3-9b89bf858891.png)

andrewtkh1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,151 @@
+* `makeMemberEntry()`: Creates a member in the memberList.
+* `writeMemberFile()`: Writes the data to the csv file.
+
+The *sequence diagram* below shows how various components of the architecture interact with one another when a user inputs a **valid** command `"delete /m 1"`
```
Perhaps it would be even better if you could give a couple of lines explaining how does the sequence diagram flows?

e.g,
1) User types in command `delete /m 1` and it... (Does something with the `Entry` class)
2) The `Entry` class calls the `getKeywordStatus()` method to.... (Get status?)
3) ....

Teckwhye committed:
Add assert

Teckwhye committed:
Merge pull request #131 from Teckwhye/TeckHwee-v2.0

Add assert

Teckwhye committed:
Merge pull request #132 from AY2122S1-CS2113T-F12-4/teckhwee-v2.0

Teckhwee v2.0

ianwangeeen committed:
add exception for delete attendance function

ianwangeeen committed:
add exception for delete attendance function

add exception for delete attendance function

glenn-tck committed:
Update README.md

glenn-tck committed:
Update README.md

glenn-tck committed:
Create README.md

Commit:
Add training storage functionality

glenn-tck committed:
Merge pull request #134 from poppolette/glenn-v2.0

Add training storage functionality

glenn-tck committed:
Update UserGuide.md

ianwangeeen committed:
add documentation and tests

ianwangeeen committed:
add documentation and tests

ianwangeeen committed:
fix test case on startup

ianwangeeen committed:
fix test case on startup

fix test case on startup

ianwangeeen committed:
add documentation and tests

add documentation and tests

izdiyadfrhn committed:
Update UserGuide.md

izdiyadfrhn committed:
Update UserGuide.md

izdiyadfrhn committed:
Update UserGuide.md

izdiyadfrhn committed:
Update UserGuide.md

izdiyadfrhn committed:
Update UserGuide.md

izdiyadfrhn committed:
Update UserGuide.md

glenn-tck committed:
Update UserGuide.md

Commit:
Update DG

glenn-tck committed:
Merge pull request #139 from poppolette/glenn-v2.0

Update DG

Commit:
Update DG

glenn-tck committed:
Merge pull request #140 from poppolette/glenn-v2.0

Update DG

Commit:
Update DG

glenn-tck committed:
Merge pull request #141 from poppolette/glenn-v2.0

Update DG

glenn-tck committed:
Update DeveloperGuide.md

xingyuan123 committed:
reposense

xingyuan123 committed:
m

xingyuan123 committed:
m

xingyuan123 committed:
sus

xingyuan123 committed:
Merge pull request #144 from xingyuan123/master

trying to solve sync error

izdiyadfrhn committed:
Update UserGuide.md

Closes #201

izdiyadfrhn committed:
fixed user guide and cleaned up some

izdiyadfrhn committed:
fixed incorrect print of empty attendance list message

izdiyadfrhn committed:
Merge pull request #207 from izdiyadfrhn/izdiyad-v2.1

Izdiyad v2.1

izdiyadfrhn committed:
Handled improper inputs and added appropriate error message when displaying full attendance list

izdiyadfrhn committed:
Merge pull request #208 from izdiyadfrhn/izdiyad-v2.1

Handled improper inputs and added appropriate error message when displaying full attendance list

izdiyadfrhn committed:
Can only add attendance if member and training exists (has been added into CCA manager beforehand)

izdiyadfrhn committed:
fix coding style

izdiyadfrhn committed:
Merge pull request #209 from izdiyadfrhn/izdiyad-v2.1

Can only add attendance if member and training exists (has been added into CCA manager beforehand)

izdiyadfrhn committed:
Handle exception of adding duplicate attendance entries. Once a particular matching member and training is an entry, cannot be added with a different attendance status anymore.

izdiyadfrhn committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F12-4/tp into izdiyad-v2.1

# Conflicts:
#	src/main/java/seedu/duke/command/AddAttendance.java

izdiyadfrhn committed:
Merge pull request #210 from izdiyadfrhn/izdiyad-v2.1

Izdiyad v2.1

ianwangeeen committed:
change UG

ianwangeeen committed:
Merge pull request #211 from ianwangeeen/master

change UG

Commit:
Fix coding errors

glenn-tck committed:
Merge branch 'master' into Glenn-v2.1

glenn-tck committed:
Merge pull request #212 from poppolette/Glenn-v2.1

Fix coding errors

ianwangeeen committed:
improve OOP for Parser class and add 'find' to help message

ianwangeeen committed:
improve OOP for Parser class and add 'find' to help message

improve OOP for Parser class and add 'find' to help message

Commit:
Update UG

Commit:
Update UG

Commit:
Update UG

Commit:
Fix codingstyle error

Commit:
Fix test cases

Commit:
Fix text-ui-test

Commit:
Fix text-ui-test

Commit:
Fix text-ui-test ????

Commit:
fix test

glenn-tck committed:
Merge pull request #214 from poppolette/Glenn-v2.1

Update UG

Commit:
Add glenn PPP

glenn-tck committed:
Merge branch 'master' into Glenn-v2.1

glenn-tck committed:
Merge pull request #215 from poppolette/Glenn-v2.1

Add glenn PPP and refactored code

izdiyadfrhn committed:
training name in attendance toUpperCase and set up my PPP

xingyuan123 committed:
java docs comments

izdiyadfrhn committed:
Updated UserGuide in Attendance

izdiyadfrhn committed:
Update MemberStorage.java

izdiyadfrhn committed:
Update AttendanceStorage.java

xingyuan123 committed:
Merge pull request #216 from xingyuan123/backup

added java docs comments to member & attendance storage

xingyuan123 committed:
Merge pull request #217 from izdiyadfrhn/izdiyad-v2.1

Izdiyad v2.1

ianwangeeen committed:
add more exceptions for edit member and MemberStorage

ianwangeeen committed:
Merge branch 'master' into master

ianwangeeen committed:
add more exceptions for edit member and MemberStorage

add more exceptions for edit member and MemberStorage

ianwangeeen committed:
Update UserGuide.md

xingyuan123 committed:
add member checks

xingyuan123 committed:
Merge pull request #219 from xingyuan123/backup

add member checks for add member

xingyuan123 committed:
no duplicates for member

xingyuan123 committed:
Merge pull request #220 from xingyuan123/backup

no duplicates for member adding

xingyuan123 committed:
issue 186

xingyuan123 committed:
Merge pull request #221 from xingyuan123/master

fixes issue 186

xingyuan123 committed:
issue 147

xingyuan123 committed:
Merge pull request #222 from xingyuan123/master

nothing major

izdiyadfrhn committed:
Attendance portion of UG completed and cleaned up some code

xingyuan123 committed:
member details verified

izdiyadfrhn committed:
Merge branch 'master' into izdiyad-v2.1

xingyuan123 committed:
member duplicates from excel file

xingyuan123 committed:
training schedule excel done

Teckwhye committed:
Fix validation bugs and exception handlings

ianwangeeen committed:
Add object diagrams for Entry, Parser, Command, UI and change seq diag to proper user

ianwangeeen committed:
fix formatting issues

ianwangeeen committed:
fix minor formatting probs

ianwangeeen committed:
fix minor formats

ianwangeeen committed:
add obj diagrams (Entry, Parser, Command, Ui) to UG and ian's PPP 

 add obj diagrams (Entry, Parser, Command, Ui) to UG and ian's PPP

ianwangeeen committed:
add obj diagram to DG (Ui, Parser, Entry, Command) and ian's PPP

add obj diagram to DG (Ui, Parser, Entry, Command) and ian's PPP

xingyuan123 committed:
gradle

xingyuan123 committed:
Merge branch 'master' into master

xingyuan123 committed:
Merge pull request #231 from xingyuan123/master

gradle

xingyuan123 committed:
gradle

xingyuan123 committed:
Merge branch 'master' of https://github.com/xingyuan123/tp

xingyuan123 committed:
Merge pull request #232 from xingyuan123/master

gradle again

izdiyadfrhn committed:
Merge branch 'master' into izdiyad-v2.1

izdiyadfrhn committed:
Update MemberStorage.java

izdiyadfrhn committed:
Create MemberStorage.java

izdiyadfrhn committed:
Merge pull request #223 from izdiyadfrhn/izdiyad-v2.1

Attendance portion of UG completed and cleaned up some code

izdiyadfrhn committed:
Update perform IO redirection test to pass

izdiyadfrhn committed:
Merge pull request #233 from izdiyadfrhn/izdiyad-v2.1

Update perform IO redirection test to pass

izdiyadfrhn committed:
Update DeveloperGuide.md

Teckwhye committed:
Edit to follow coding style

Teckwhye committed:
Refactor to follow coding style

Teckwhye committed:
Merge pull request #225 from Teckwhye/TeckHwee-v2.1

Teck hwee v2.1

Teckwhye committed:
Merge branch 'master' into teckhwee-v2.0

Teckwhye committed:
Fix edit function

Teckwhye committed:
Fix JUnit test

Teckwhye committed:
Merge pull request #235 from Teckwhye/TeckHwee-v2.1

Fix edit function

Teckwhye committed:
Merge pull request #234 from AY2122S1-CS2113T-F12-4/teckhwee-v2.0

Teckhwee v2.1 bug fix

Commit:
Update UG

glenn-tck committed:
Merge branch 'master' into Glenn-v2.1

glenn-tck committed:
Merge pull request #236 from poppolette/Glenn-v2.1

Update UG, fix some bugs

Commit:
Fix codingstyle error

Commit:
Fix codingstyle error

glenn-tck committed:
Merge pull request #237 from poppolette/Glenn-v2.1

Fix codingstyle error

Commit:
add authortag for poppolette

glenn-tck committed:
Merge pull request #238 from poppolette/Glenn-v2.1

add authortag for poppolette

glenn-tck committed:
Update UserGuide.md

glenn-tck committed:
Update UserGuide.md

Teckwhye committed:
Add exception handling and PPP

Teckwhye committed:
Add author tag to DG

Teckwhye committed:
Refactor to meet coding styte

Teckwhye committed:
Merge pull request #239 from Teckwhye/TeckHwee-v2.1.2

Teck hwee v2.1.2

Teckwhye committed:
Merge pull request #240 from AY2122S1-CS2113T-F12-4/teckhwee-v2.0

Teckhwee v2.1

Commit:
Add more JUnit test cases for TrainingList

Commit:
Fix checkstyle error

Commit:
Fix checkstyle error

Commit:
Fix checkstyle error

glenn-tck committed:
Merge pull request #241 from poppolette/Glenn-v2.1

Add more JUnit test cases for TrainingList

xingyuan123 committed:
attendance storage

xingyuan123 committed:
Merge pull request #242 from xingyuan123/master

attendance storage

ianwangeeen committed:
resolve UG member section and Acknowledgements

ianwangeeen committed:
resolve UG member section and Acknowledgements

ianwangeeen committed:
Merge pull request #244 from ianwangeeen/master

resolve UG member section and Acknowledgements

ianwangeeen committed:
resolve UG member section and Acknowledgements

resolve UG member section and Acknowledgements

ianwangeeen committed:
Update DeveloperGuide.md

ianwangeeen committed:
Update DeveloperGuide.md

ianwangeeen committed:
Update DeveloperGuide.md

ianwangeeen committed:
Update DeveloperGuide.md

ianwangeeen committed:
Update DeveloperGuide.md

ianwangeeen committed:
Update DeveloperGuide.md

ianwangeeen committed:
Update UserGuide.md

ianwangeeen committed:
Update UserGuide.md

ianwangeeen committed:
Update UserGuide.md

ianwangeeen committed:
Update UserGuide.md

ianwangeeen committed:
Update UserGuide.md

ianwangeeen committed:
Update UserGuide.md

ianwangeeen committed:
Update UserGuide.md

ianwangeeen committed:
Update UserGuide.md

ianwangeeen committed:
Update UserGuide.md

ianwangeeen committed:
Update UserGuide.md

ianwangeeen committed:
Update UserGuide.md

Commit:
Fix DG architecture sequence diagram, fix DG formatting

glenn-tck committed:
Merge pull request #246 from poppolette/Glenn-v2.1

Fix DG architecture sequence diagram, fix DG formatting

Commit:
fix architecture sequence diagram

glenn-tck committed:
Merge pull request #247 from poppolette/Glenn-v2.1

fix architecture sequence diagram

izdiyadfrhn committed:
Fixed delete attendance function

izdiyadfrhn committed:
Update AttendanceStorage.java

izdiyadfrhn committed:
Update AttendanceStorage.java

izdiyadfrhn committed:
Merge pull request #248 from izdiyadfrhn/izdiyad-v2.1

Fixed delete attendance function

xingyuan123 committed:
debug and ug for storage

xingyuan123 committed:
Merge pull request #249 from xingyuan123/master

debug and ug for storage

Teckwhye committed:
Create UserGuide.md

Teckwhye committed:
Added Xing Yuan changes because I overwritten it

xingyuan123 committed:
gradle

xingyuan123 committed:
Merge pull request #250 from xingyuan123/master

gradle

izdiyadfrhn committed:
Add more JUnit test cases for AttendanceList and javadocs

izdiyadfrhn committed:
Merge pull request #251 from izdiyadfrhn/izdiyad-v2.1

Add more JUnit test cases for AttendanceList and javadocs

izdiyadfrhn committed:
Update izdiyadfrhn.md

izdiyadfrhn committed:
Update AboutUs.md

izdiyadfrhn committed:
Update README.md

xingyuan123 committed:
UG and PPP for xing yuan

xingyuan123 committed:
UG and PPP for xing yuan

xingyuan123 committed:
Merge pull request #252 from xingyuan123/master

ppp done and UG updated foe xing yuan

Teckwhye committed:
Update Test case for Member

xingyuan123 committed:
PPP for xing yuan

xingyuan123 committed:
Merge pull request #254 from xingyuan123/master

PPP for xing yuan

Teckwhye committed:
Updated values

Teckwhye committed:
Updated values for MemberTest

Teckwhye committed:
Merge pull request #253 from Teckwhye/TeckHwee-MoreTestCase

Update Test case for Member

izdiyadfrhn committed:
Update AboutUs.md

Teckwhye committed:
Add member PlantUML

Teckwhye committed:
Update MemberComponent.puml

ianwangeeen committed:
Update UserGuide.md

Commit:
Update DG diagrams

glenn-tck committed:
Merge pull request #255 from poppolette/Glenn-v2.1

v2.1 Update DG diagrams

Teckwhye committed:
Update DG

Teckwhye committed:
Merge branch 'master' into TeckHwee-MoreTestCase

Teckwhye committed:
Merge pull request #256 from Teckwhye/TeckHwee-MoreTestCase

Update DG

ianwangeeen committed:
change PPP for ian

ianwangeeen committed:
change PPP for ian

change PPP for ian

Teckwhye committed:
Edit Image

ianwangeeen committed:
change PPP for ian

change PPP for ian

Teckwhye committed:
Merge pull request #259 from Teckwhye/master

Edit Image

Commit:
Update DG and PPP

glenn-tck committed:
Merge pull request #260 from poppolette/Glenn-v2.1

v2.1 Glenn Update DG and PPP

ianwangeeen committed:
fix minor bug in parser

ianwangeeen committed:
fix minor bug in parser

fix minor bug in parser

Commit:
Update PPP

glenn-tck committed:
Merge pull request #262 from poppolette/Glenn-v2.1

v2.1 Update PPP

glenn-tck committed:
Update README.md

Commit:
Update DG

glenn-tck committed:
Merge pull request #263 from poppolette/Glenn-v2.1

Update DG

izdiyadfrhn committed:
Updated DG

izdiyadfrhn committed:
Merge pull request #264 from izdiyadfrhn/izdiyad-v2.1

Updated DG

xingyuan123 committed:
ppp

xingyuan123 committed:
ppp

xingyuan123 committed:
Merge pull request #265 from xingyuan123/master

ppp

glenn-tck committed:
Update UserGuide.md

glenn-tck committed:
Update DeveloperGuide.md

glenn-tck committed:
Update UserGuide.md
